src/ophys_etl/modules/suite2p_registration/utils.py

```diff
@@ -63,3 +64,52 @@ def identify_and_clip_outliers(data: np.ndarray,
                             data_filtered[indices] - thresh,
                             data_filtered[indices] + thresh)
     return data, indices
+
+
+def check_movie_against_raw(corr_data, raw_hdf5, h5py_key, batch_size=2000):
+    """Check that the values stored in all frames match between motion
+    corrected and raw movies.
+
+    Test is not run if the suite2p option nonrigid is used as this option
+    does not preserve flux.
+
+    Parameters
+    ----------
+    corr_data : numpy.ndarray, (N, M, L)
+        Motion corrected movie.
+    raw_hdf5 : str
+        Full path location to raw movie.
+    h5py_key : str
+        Name of the dataset containing the raw data.
+    batch_size : int
+        Number of frames to load from the h5py at once.
+    """
+    with h5py.File(raw_hdf5) as raw_file:
+        raw_dataset = raw_file[h5py_key]
+        tot_corr_frames = corr_data.shape[0]
+        tot_raw_frames = raw_dataset.shape[0]
+        if tot_corr_frames != tot_raw_frames:
+            raise ValueError('Length of motion corrected movie does '
+                             'not match length of raw movie. Something went '
+                             'wrong during motion correction. Exiting.')
+        # Loop through each chunk of data.
+        for start_idx in np.arange(
+                0, tot_corr_frames, batch_size, dtype=int):
+            end_idx = start_idx + batch_size
+            if end_idx > tot_corr_frames:
+                end_idx = tot_corr_frames
+            raw_frames = raw_dataset[start_idx:end_idx]
+            corr_frames = corr_data[start_idx:end_idx]
+            # Test each frames values.
+            for frame_idx, (raw_frame, corr_frame) in enumerate(zip(
+                    raw_frames, corr_frames)):
+                # Flatten and sort to test all array values are the same.
+                flat_raw = raw_frame.flatten()
+                flat_corr = corr_frame.flatten()
+                flat_raw.sort()
+                flat_corr.sort()
+                if not np.array_equal(flat_raw, flat_corr):
+                    raise ValueError(
+                        f'Frame {frame_idx} does not match between raw and '
```

Comment:
Can you make the error message a little more explanatory. Something like

"The distribution of pixel values in the raw movie does not match the distribution of pixel values in the corrected movie...."

Just trying to avoid the "but we don't *expect* the frames to match; that's why we're motion correcting them" confusion.